src/authutils/user.py

```diff
@@ -11,7 +11,12 @@
 
 
 def set_current_user(**kwargs):
-    flask.g.user = CurrentUser(**kwargs)
+    expected_audience = flask.current_app.config.get("USER_API")
+    # Gen3 services use both USER_API and BASE_URL
+    if not expected_audience:
+        expected_audience = flask.current_app.config.get("BASE_URL")
+
+    flask.g.user = CurrentUser(jwt_kwargs={"audience": expected_audience}, **kwargs)
```

Comment:
i don't see this function being used in many places, but theoretically i think this would error or behave in an unexpected way if `jwt_kwargs` is in `kwargs`. Or if `kwargs["jwt_kwargs"]` contains a different `audience`?

Reply:
Good point!!! For the record, yes, this
```
kwargs["jwt_kwargs"] = {"audience": "paulines_sneaky_aud"}
flask.g.user = CurrentUser(jwt_kwargs={"audience": expected_audience}, **kwargs)
```
results in this
`TypeError: type object got multiple values for keyword argument 'jwt_kwargs'`

I vill fixits; many thanks 🌻